internal/failure/failure.go

```diff
@@ -0,0 +1,268 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package failure
+
+import (
+	"fmt"
+	"sort"
+	"strconv"
+	"strings"
+	"text/scanner"
+)
+
+type (
+	// Field references a Failure field.
+	Field int
+
+	// ID references a Failure identifier.
+	ID int
+)
+
+const (
+	// Filename references the Filename field of a Failure.
+	Filename Field = iota
+	// Line references the Line field of a Failure.
+	Line
+	// Column references the Column field of a Failure.
+	Column
+	// Identifier references the ID field of a Failure.
+	Identifier
+	// Message references the Message field of a Failure.
+	Message
+
+	// Error identifies a generic Failure.
+	Error ID = iota
+	// Format identifies a diff format Failure.
+	Format
+	// Lint identifies a lint Failure.
+	Lint
+	// Proto identitifes an invalid proto Failure.
+	Proto
+
+	// DefaultFilename is the default Failure filename.
+	DefaultFilename = "<input>"
+	// DefaultLine is the default Failure line.
+	DefaultLine = 1
+	// DefaultColumn is the default Failure column number.
+	DefaultColumn = 1
+)
+
+var (
+	_defaultFields = []Field{
+		Filename,
+		Line,
+		Column,
+		Message,
+	}
+	_stringToField = map[string]Field{
+		"filename": Filename,
+		"line":     Line,
+		"column":   Column,
+		"id":       Identifier,
+		"message":  Message,
+	}
+	_IDToString = map[ID]string{
```

Comment:
What is this doing specifically? The ID is primarily meant to reference lint IDs, which these are not. If you were to enumerate all possible IDs, this would be in the `lint` package, which should not be a dependency of this package.

Reply:
Look at `internal/x/lint/base_checker.go`, function `baseChecker#Check`. The issue here is that the concept of ID is getting overloaded. We should either have a lint ID, or not. If anything, we should remove `INVALID_PROTOBUF`, etc.